src/ports/postgres/modules/convex/keras_model_arch_table.py_in

```diff
@@ -0,0 +1,133 @@
+    if not table_exists(model_arch_table):
+        col_defs = ','.join(map(' '.join,
+                            zip(Format.col_names,
+                                Format.col_types)))
+
+        sql = "CREATE TABLE {model_arch_table} ({col_defs})" \
+              .format(**locals())
+
+        _execute(sql)
+        plpy.info("Created new keras model arch table {0}." \
+            .format(model_arch_table))
+    else:
+        missing_cols = columns_missing_from_table(model_arch_table, Format.col_names)
+        if len(missing_cols) > 0:
+            plpy.error("Invalid keras model arch table {0},"
+                       " missing columns: {1}".format(model_arch_table, missing_cols))
+
+    unique_str = unique_string(prefix_has_temp=False)
+
+    sql = """INSERT INTO {model_arch_table} ({model_arch_col}, {internal_id_col})
+                                    VALUES({model_arch}, '{unique_str}');
+             SELECT model_id, model_arch
+                 FROM {model_arch_table} WHERE {internal_id_col} = '{unique_str}'
+    """.format(model_arch_table=model_arch_table,
+               model_arch_col=Format.model_arch,
+               unique_str=unique_str,
+               model_arch=quote_literal(model_arch),
+               internal_id_col=Format.__internal_madlib_id__)
+
+# This code works perfectly in postgres 8.3+, but fails in Greenplum 5 with:
+#   ERROR:  The RETURNING clause of the INSERT statement is not supported in this version
+#           of Greenplum Database.
+#    sql = """INSERT INTO {model_arch_table}
+#        (model_arch)
+#        VALUES('{model_arch}')
+#        RETURNING *;""".format(model_arch_table=model_arch_table,
+#                                model_arch=quote_literal(model_arch))
+    res = _execute(sql,1)
+
+    if len(res) != 1 or res[0][Format.model_arch] != model_arch:
+        raise Exception("Failed to insert new row in {0} table--try again?"
+                       .format(model_arch_table))
+    plpy.info("Added model id {0} to {1} table".format(res[0]['model_id'], model_arch_table))
+
+def delete_keras_model(keras_model_arch_table, model_id, **kwargs):
+    model_arch_table = quote_ident(keras_model_arch_table)
+    if not table_exists(model_arch_table):
+        plpy.error("Table {0} does not exist.".format(model_arch_table))
+
+    missing_cols = columns_missing_from_table(model_arch_table, Format.col_names)
+    if len(missing_cols) > 0:
+        plpy.error("Invalid keras model arch table {0},"
+                   " missing columns: {1}".format(model_arch_table, missing_cols))
+
+    sql = """
+           DELETE FROM {model_arch_table} WHERE model_id={model_id}
+          """.format(model_arch_table=model_arch_table, model_id=model_id)
+    res = _execute(sql)
+
+    if res.nrows() > 0:
+        plpy.info("Model id {0} has been deleted from {1}.".format(model_id, model_arch_table))
+    else:
+        plpy.info("Model id {0} not found".format(model_id))
```

Comment:
Should this be `plpy.error()` instead of `info`?

Reply:
I don't feel strongly either way, but we should discuss as a group and decide.  I think the question boils down to whether we want it to act more like "DROP" or "DROP IF EXISTS", both of which can be useful at times.

Reply:
Oh, actually maybe plpy.warn()?  I think I like that more than error or info... warn the user, but don't throw an exception.